vcd/datasource_vcd_org_vdc_test.go

```diff
@@ -0,0 +1,115 @@
+// +build vdc ALL functional
+
+package vcd
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+func TestAccVcdVdcDatasource(t *testing.T) {
```

Comment:
I get the impression, that this test does not test the data source and its fields, but vice versa. That is - it uses a data source to generate some vdc config and checks that the resource is provisioned as specified from the data source. Wondering if this can mask some issues with data source itself (for example if it does not retrieve some fields correctly but resource provisioning still works)

Reply:
This test according Giuseppes recommendations, his done previously his investigation. It's best approach though it has own issues.

Reply:
Sorry don't get idea - what is difference if I would check : resource.TestCheckResourceAttr("vcd_org_vdc."+vdcName, "provider_vdc_name", testConfig.VCD.ProviderVdc.Name)
Why we would need output?

Reply:
Your approach is equivalent. 
Using `output` shows explicitly that we are accessing the data source in the HCL script.